### PR TITLE
[Rom] Wrote script to visualize basis functions

### PR DESCRIPTION
**Pull request by EduardGomezEscandell**
**📝 Description**
It's an analysis stage which instead of solving, it reads the `RomParameters.json` and creates a GiD output for the basis.

For simplicity-of-implementation's sake, it assigns each row of the basis matrix to a variable, and each column to a diferent time-step.

For instance, with thermal unit test `mdpa` and `RomParameters.json`, plus these parameters:
```
{
    "solver_settings": {
        "model_part_name": "main_model_part", // Doesn't matter what you put here
        "model_import_settings": {
            "input_type": "mdpa",
            "input_filename": "Square_Radiation_Stationary"
        },
        "rom_parameters_file" : "RomParameters.json"
    }
}
```
the produced output for the first basis looks like:
image

The basis is assigned to the variable `TEMPERATURE` because it is the only variable specified in `RomParameters.json`.

**🆕 Changelog**
- Created `BasisVisualizationAnalysis`
- Created unit test

### CODE REVIEW

File: applications/RomApplication/python_scripts/basis_visualization_analysis.py [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+        settings_filename = self.settings["rom_parameters_file"].GetString()
+        with open(settings_filename, "r") as f:
+            rom_params = KratosMultiphysics.Parameters(f.read())
+
+        self.rom_settings = rom_params
+
+        self.number_of_modes = rom_params["rom_settings"]["number_of_rom_dofs"].GetInt()
+
+        self.variable_names = [p.GetString() for p in rom_params["rom_settings"]["nodal_unknowns"]]
+        self.variables = None
```
Comment (philbucher) [inline applications/RomApplication/python_scripts/basis_visualization_analysis.py]:
```suggestion
        self.variable_names = rom_params["rom_settings"].GetStringArray()
        self.variables = kratos_utilities.GenerateVariableListFromInput(rom_params["rom_settings"])
```

https://github.com/KratosMultiphysics/Kratos/blob/master/kratos/python_scripts/kratos_utilities.py

**Comment by emiroglu** [comment]
Hi @EduardGomezEscandell , there has already been a similar implementation here, which has not made yet to the main branch: https://github.com/KratosMultiphysics/Kratos/tree/IMWind/applications/RomApplication/custom_processes

You could take a look in case you need inspiration.

**Comment by emiroglu** [comment]
I am actually trying to get #8174 through from quite some time ago. It contains also some output related functions and processes for both GiD and Paraview:
https://github.com/KratosMultiphysics/Kratos/tree/RomApp_ModalDerivatives_ModalParameters/applications/RomApplication/custom_io
https://github.com/KratosMultiphysics/Kratos/tree/RomApp_ModalDerivatives_ModalParameters/applications/RomApplication/custom_processes

**Comment by EduardGomezEscandell** [comment]
Closing because there are cleaner solutions (Using an output process).

**Pull request closed without merging**